openedx/core/djangoapps/profile_images/views.py

```diff
@@ -15,10 +17,22 @@
     SessionAuthenticationAllowInactiveUser,
 )
 from openedx.core.lib.api.permissions import IsUserInUrl, IsUserInUrlOrStaff
-from openedx.core.djangoapps.user_api.accounts.image_helpers import set_has_profile_image, get_profile_image_names
+from openedx.core.djangoapps.user_api.accounts.image_helpers import (
+    get_profile_image_version,
+    set_profile_image_version,
+    get_profile_image_names,
+)
 from .images import validate_uploaded_image, create_profile_images, remove_profile_images, ImageValidationError
 
 
+def _make_image_version():
+    """
+    Generate a version based on hashing timestamp.  This is in a separate function
+    so its behavior can be overridden in tests.
+    """
+    return hashlib.md5(str(time.time())).hexdigest()[:8]
```

Comment:
the value only needs to be unique per-user.  the probability (i checked!) of a user getting the same version stamp twice in 1,000 uploads, under the above function, is 0.0001, so it's just to shave off information we don't need and make urls / versions shorter.